modules/editor/grammar/post.pegjs

```diff
@@ -0,0 +1,95 @@
+Document
+  = WP_Block_List
+
+WP_Block_List
+  = WP_Block*
+
+WP_Block
+  = WP_Block_Balanced
+  / WP_Block_Html
+
+WP_Block_Balanced
+  = s:WP_Block_Start ts:(!WP_Block_End c:Any { return c })+ e:WP_Block_End
+  { return {
+    blockType: s.blockType,
+    attrs: s.attrs,
+    rawContent: ts.join( '' ),
+  } }
+
+WP_Block_Html
+  = ts:(!WP_Block_Balanced c:Any { return c })+
+  {
+    return {
+      blockType: 'html',
+      attrs: {},
+      rawContent: ts.join('')
+    }
+  }
+
+WP_Block_Start
+  = "<!--" __ "wp:" blockType:WP_Block_Type attrs:WP_Block_Attribute_List _? "-->"
+  { return {
+    type: 'WP_Block_Start',
+    blockType,
+    attrs,
+    text: text()
+  } }
+
+WP_Block_End
+  = "<!--" __ "/wp" __ "-->"
```

Comment:
It doesn't "need" to be there but would definitely should be. It will make it easy for human eyeballs rather than computer eyeballs. Imagine trying to read html or xml without a matching closing tag, a parser could technically understand it but it would be a lot less friendly to work with.

I think the points nylen brought up are all very compelling as well.